runtime-modules/bounty/src/lib.rs

```diff
@@ -1118,127 +1377,210 @@ decl_module! {
 
+            Self::ensure_work_entry_ownership(&entry, &member_id)?;
+
             //
             // == MUTATION SAFE ==
             //
 
-            // Claim the winner reward.
-            if let Some(OracleWorkEntryJudgment::Winner { reward }) = entry.oracle_judgment_result {
-                Self::transfer_funds_from_bounty_account(
-                    &controller_account_id,
+            T::StakingHandler::unlock(&entry.staking_account_id);
+            Self::deposit_event(
+                RawEvent::WorkEntrantStakeUnlocked(
                     bounty_id,
-                    reward
-                )?;
+                    entry_id,
+                    entry.staking_account_id)
+            );
+            Self::remove_work_entry(&bounty_id, &entry_id);
+
+            if Self::withdrawal_completed(&current_bounty_stage, &bounty_id) {
+                Self::remove_bounty(&bounty_id);
             }
+        }
 
-            // Unstake the full work entry state.
-            T::StakingHandler::unlock(&entry.staking_account_id);
+        ///Withraws the state bloat bond to funder
+        ///If bounty is successfully, funders must call this extrinsic to withdraw the state bloat bond,
+        ///this makes the cleaning storage process more efficient.
+        /// # <weight>
+        ///
+        /// ## weight
+        /// `O (1)`
+        /// - db:
+        ///    - `O(1)` doesn't depend on the state or parameters
+        /// # </weight>
+        #[weight = WeightInfoBounty::<T>::withdraw_state_bloat_bond_by_council()
+        .max(WeightInfoBounty::<T>::withdraw_state_bloat_bond_by_member())]
+        pub fn withdraw_state_bloat_bond(
+            origin,
+            funder: BountyActor<MemberId<T>>,
+            bounty_id: T::BountyId,
+        ) {
+            let bounty_funder_manager = BountyActorManager::<T>::ensure_bounty_actor_manager(
+                origin,
+                funder.clone(),
+            )?;
 
-            // Delete the work entry record from the storage.
-            Self::remove_work_entry(&bounty_id, &entry_id);
+            let bounty = Self::ensure_bounty_exists(&bounty_id)?;
+
+            let current_bounty_stage = Self::get_bounty_stage(&bounty);
+
+            ensure!(current_bounty_stage == BountyStage::SuccessfulBountyWithdrawal,
+                Self::unexpected_bounty_stage_error(current_bounty_stage)
+            );
+            ensure!(
+                <BountyContributions<T>>::contains_key(&bounty_id, &funder),
+                Error::<T>::NoBountyContributionFound,
+            );
+
+            let funding = <BountyContributions<T>>::get(&bounty_id, &funder);
+            let account_id = Self::bounty_account_id(bounty_id);
+
+            ensure!(Self::check_balance_for_account(funding.state_bloat_bond, &account_id),
+                Error::<T>::NoBountyBalanceToStateBloatBondWithdrawal);
+
+            //
+            // == MUTATION SAFE ==
+            //
+
```

Comment:
I believe that reason we don't need to care about cherry & oracle reward payout is because it has already been done during judgement?